Core/include/Acts/EventData/detail/Charge.hpp

```diff
@@ -49,28 +50,38 @@ namespace Acts {
 ///
 /// @{
 
+struct Neutral;
+struct SinglyCharged;
+class AnyOrNoCharge;
+class AnyButNoCharge;
+
 /// Charge and momentum interpretation for neutral particles.
 struct Neutral {
   Neutral() = default;
   /// Construct and verify the input charge magnitude (in debug builds).
   ///
   /// This constructor is only provided to allow consistent construction.
-  template <typename T>
-  constexpr Neutral(T absQ) noexcept {
-    assert((absQ == static_cast<T>(0)) and "Input charge must be zero");
+  constexpr Neutral(float absQ) noexcept {
+    assert((absQ == 0) and "Input charge must be zero");
     // suppress `unused variable` warning in non-debug builds
     (void)(absQ);
```

Comment:
Just as a question: does the `[[maybe_unused]]` attribute not work for some reason? I've seen these `void`-casts quite a lot lately and wonder why we don't use the attribute...